### PR TITLE
Members-Only: Ruby on Rails project based on Authentication.

### PR DESCRIPTION
**Pull request by vicmaburrito**
In this app project:
- After the initial commit, we followed the git workflow and set up the .github/workflows linters file.
- Then we set up local env. linters with rubocop and stylelint
- We then elaborate authentications for the member so that they can have access to create, delete and modify post as registered member only, If you are not a registered user, you can only look at the publications but without knowing who the author was or any reference to posts.
- Fix all linters errors
- Then Add proper documentation to our ReadMe.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
+  root "posts#index"
+  devise_for :members, :controllers => { registrations: 'registrations' }
+  resources :posts
```
Comment (rindrajosia) [inline config/routes.rb:5]: It creates the application routes using `resources :posts, only: [:new, :create, :index]`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_member!, except: %i[index new create]
```
Comment (rindrajosia) [inline app/controllers/posts_controller.rb]: before_action :authenticate_member!, only: [:new, :create]

File: README.md [hunk 3]
```diff
@@ -1,24 +1,56 @@
...
 
-* System dependencies
 
-* Configuration
+## Usage 
 
-* Database creation
+- Clone the repo: $ git clone
+- Run `bundle install`  in your terminal to install the Ruby gems.
+- Run `rails db:migrate`  in your terminal to create the databases.
```
Comment (rindrajosia) [inline README.md]: Please ask the user to create DB via r`ails db:create` before migration

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- User can sign in :+1: 
- User you can create new post :+1: 
- User can sign_out :+1: 

To improve :factory_worker: [REQUIRED]:

- [x] It creates the application routes using `resources :posts, only: [:new, :create, :index] `([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- [x] it uses before_action to restrict access to `the :new and :create` methods ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

Nice to have [OPTIONAL] :

- [x] Please move logic in your view and partial inside helpers

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

File: config/routes.rb (hunk 1 above)
Comment (DanSam5K) [reply config/routes.rb:5]: Noted

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (DanSam5K) [reply app/controllers/posts_controller.rb]: Noted!

Comment (DanSam5K) [reply app/controllers/posts_controller.rb:3]: 👍

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coder, 👋🏽 
**This is a great project you have going on, well-done lads and thanks for implementing the changes requested earlier on**. 

### Well-done 💯 
    1. The app runs as expected.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] Check inline comments for sections to improve

### OPTIONAL

- [x] Remove all pieces of commented-out code.

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  before_action :authenticate_member!, only: %i[new create]
```
Comment (mke2111) [inline app/controllers/posts_controller.rb:3]: - [x] You currently have a conflict with the access rights using `before_action`. I suggest you keep the second `before_action` as previously requested.

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  root "posts#index"
+  devise_for :members, :controllers => { registrations: 'registrations' }
+  # resources :posts
```
Comment (mke2111) [inline config/routes.rb]: - [x] [OPTIONAL] Remove this commented-out code

Comment (DanSam5K) [reply config/routes.rb]: Alright, Noted thank you

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hello Team 👋🏽👋🏽👋🏽
Your project is really nice!

📝 Please help me to make this simple changes:
- Add instructions about Optional installations like: ```rails webpacker:install``` 
- Kindly remove unused files like: ```.json.jbuilder```
- After create a Post, I have this error message: 
Screen Shot 2021-06-17 at 05 56 19


_Happy Coding!_ 👨🏽‍💻🥷🏽👓

@Carlos-ssh

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔**

Great work making the required changes.👏👏👏
You can merge and move on👍👍👍

Happy coding💻💻

**Pull request merged**